File: 문제풀이/chap3-2/prob3-2-3.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		// 입력
		int N = sc.nextInt();
		long[] A = new long[N + 1];
		for (int i = 1; i <= N; i++) A[i] = sc.nextLong();
		
		// 답 구하기
		long R = LCM(A[1], A[2]);
		for (int i = 3; i <= N; i++) {
			R = LCM(R, A[i]);
		}
		
		// 출력
		System.out.println(R);
	}
	static long GCD(long A, long B) {
		// 최대공약수를 리턴하는 함수
		while (A >= 1 && B >= 1) {
			if (A < B) {
				B %= A;
			}
			else {
				A %= B;
			}
		}
		if (A >= 1) {
			return A;
		}
		return B;
	}
	static long LCM(long A, long B) {
		// 최소공배수를 리턴하는 함수
		return (A / GCD(A, B)) * B;
	}
}

File: 문제풀이/chap3-2/prob3-2-3.py
# 최대공약수를 리턴하는 함수
def GCD(A, B):
	while A >= 1 and B >= 1:
		if A < B:
			B = B % A  # A < B라면, 큰 수를 b로 변경합니다.
		else:
			A = A % B  # A >= B라면 큰 수를 a로 변경합니다.
	if A >= 1:
		return A
	return B

# 최소공배수를 리턴하는 함수
def LCM(A, B):
	return int(A / GCD(A, B)) * B

# 입력
N = int(input())
A = list(map(int, input().split()))

# 답 구하기
R = LCM(A[0], A[1])
for i in range(2, N):
	R = LCM(R, A[i])

# 출력
print(R)